src/test/java/com/ably/kafka/connect/ConfigValueEvaluatorTest.java

```diff
@@ -91,10 +91,49 @@ void testEvaluateNullPatternReturnsNull() {
         SinkRecord sinkRecord = new SinkRecord("topic", 0, null, "key".getBytes(), null, null, 0);
 
         //when
-        String result = configValueEvaluator.evaluate(sinkRecord, null);
+        ConfigValueEvaluator.Result result = configValueEvaluator.evaluate(sinkRecord, null, false);
 
         //then
-        assertNull(result);
+        assertNull(result.getValue());
+    }
+
+    @Test
+    void testEvaluateKeyIsEvaluatedNormallyWhenSkippableRecordIsProvided() {
+        //given
+        SinkRecord sinkRecord = new SinkRecord("greatTopic", 0, null,"greatKey".getBytes(), null, null, 0);
+
+        //when
+        ConfigValueEvaluator.Result result = configValueEvaluator.evaluate(sinkRecord, "#{key}_hello", true);
+
+        //then
+        assertEquals("greatKey_hello", result.getValue());
+    }
+
+    @Test
+    void testEvaluateSkipResultIsReturnedWhenRecordWithoutKeyWasProvided() {
+        //given
+        SinkRecord sinkRecord = new SinkRecord("greatTopic", 0, null,null, null, null, 0);
+
+        //when
+        ConfigValueEvaluator.Result result = configValueEvaluator.evaluate(sinkRecord, "#{key}_hello", true);
+
+        //then
+        assertTrue(result.shouldSkip());
+        assertNull(result.getValue());
+    }
+
+    //make sure that #{topic} won't impact the behaviour
```

Comment:
Is it some kind of TODO comment or does it describe the below test case? If the latter maybe we should put this information in the test name rather than in a comment above it? 🤔

Reply:
It describes the test case indeed